File: JAVA/LEVEL1/sortString.java
/*
문자열로 구성된 리스트 strings와, 정수 n이 주어졌을 때, 각 문자열의 인덱스 n번째 글자를 기준으로 오름차순 정렬하려 합니다. 
예를 들어 strings가 ["sun", "bed", "car"]이고 n이 1이면 각 단어의 인덱스 1의 문자 "u", "e", "a"로 strings를 정렬합니다.
*/

import java.util.*;

class Solution {
    public String[] solution(String[] strings, int n) {
        Arrays.sort(strings);
        String[] answer = new String[strings.length];
        
        for(int i = 0; i < strings.length; i++){
           answer[i] = strings[i].substring(n, n+1);
        }
        Arrays.sort(answer);
        
        List<String> list = new ArrayList<>(Arrays.asList(strings));
        for(int i = 0; i < answer.length; i++){
            for(int j = 0; j < list.size(); j++){
                if(list.get(j).substring(n, n+1).equals(answer[i])){
                    answer[i] = list.get(j);
                    list.remove(list.get(j));
                    break;
                }
            }
        }
        return answer;
    }
}

File: PYTHON/LEVEL1/sortString.py
def solution(strings, n):
    list = []
    answer = []
    for i in range(len(strings)):
        list.append(strings[i][n:n+1])
    

    strings.sort()
    list.sort()
    
    for j in range(len(list)):
        for i in range(len(strings)):
            if strings[i][n:n+1] == list[j]:
                if strings[i] not in answer:
                    answer.append(strings[i])  
    return answer